### PR TITLE
Initial Support for flavors

### PR DESCRIPTION
**Pull request by edTheGuy00**
Hello,

As discussed in https://github.com/go-flutter-desktop/go-flutter/issues/475 this is my attempt to implement application flavors for Hover. I tested on my MacBook and was able to build and run different application flavors with no problem. Additional testing might be needed for other platforms which I don not have access to. 

I also believe a new docker image will need to be built in order to get this working for docker builds.

### CODE REVIEW

File: internal/config/flavor.go [hunk 1]
```diff
@@ -0,0 +1,40 @@
+package config
+
+import (
+	"github.com/go-flutter-desktop/hover/internal/build"
+	"github.com/go-flutter-desktop/hover/internal/log"
+	"os"
+	"path/filepath"
```
Comment (jld3103) [inline internal/config/flavor.go:7]:
```suggestion
	"os"
	"path/filepath"
	
	"github.com/go-flutter-desktop/hover/internal/build"
	"github.com/go-flutter-desktop/hover/internal/log"
```

File: internal/config/flavor.go [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+// SetDefaultFlavorFile sets the default hover.yaml
+func SetDefaultHoverYamlFile() {
+	hoverYaml = "hover.yaml"
+}
+
+// SetHoverFlavor sets the user defined hover flavor.
+// eg. hover-develop.yaml, hover-staging.yaml, etc.
+func SetHoverFlavor(flavor string) {
+	hoverYaml = "hover-" + flavor + ".yaml"
```
Comment (jld3103) [inline internal/config/flavor.go:25]:
```suggestion
	hoverYaml = fmt.Sprintf("hover-%s.yaml", flavor)
```

File: internal/config/flavor.go [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+// assertYamlFileExists checks to see if the user defined yaml file exists
+func assertYamlFileExists(yamlFile string) {
+	_, err := os.Stat(filepath.Join(build.BuildPath, yamlFile))
+	if os.IsNotExist(err) {
+		log.Warnf("Hover Yaml file \"%s\" not found.", yamlFile)
+		os.Exit(1)
+	}
+	if err != nil {
+		log.Errorf("Failed to stat hover.yaml: %v\n", err)
```
Comment (jld3103) [inline internal/config/flavor.go:37]:
```suggestion
		log.Errorf("Failed to stat %s: %v\n", yamlFile, err)
```

**Review by jld3103 — COMMENTED** [review]
Other than my suggestions this looks really good!
The new docker image will be built when we release a new hover version which is probably in the next days when https://github.com/go-flutter-desktop/hover/pull/170 is confirmed to fix everything.

**Comment by edTheGuy00** [comment]
@jld3103 thanks, I have updated as suggested

**Pull request merged**